### PR TITLE
fix scanner split infinite loop cause by extftool error output

### PR DESCRIPTION
**Pull request by switch-li**
if exiftool parse file error, maybe return unknow string (This is usually caused by exifTool error output).
e.g.
filehash: 8e2a7aaeeee829f77b1a1029b9f7524879bbe399
outpuut: 'x' outside of string in unpack at /usr/share/perl5/vendor_perl/Image/ExifTool.pm line 5059.

### CODE REVIEW

**Comment by switch-li** [comment]
exiftool output not always return string split by "{ready}\n"
lot of times return a error message split by "\n"

**Comment by barasher** [comment]
Hi, thanks for the PR.
Could you please add some unit tests ?

File: exiftool.go [hunk 1]
```diff
@@ -122,8 +122,8 @@ func (e *Exiftool) Close() error {
 
 	// Wait for wait to finish or timeout
 	select {
-	case <- ch:
```
Comment (barasher) [inline exiftool.go:125]: This code doesn't seem to respect golang coding style. Could you please run gofmt?

File: exiftool.go [hunk 2]
```diff
@@ -269,16 +269,29 @@ func (e *Exiftool) WriteMetadata(fileMetadata []FileMetadata) {
...
-			return 0, data, fmt.Errorf("no final token found")
+	if atEOF && len(data) == 0 {
+		return 0, nil, nil
+	}
+
+	// if exiftool parse file error, maybe return unknow string (This is usually caused by exifTool error output).
+	// e.g.
+	// filehash: 8e2a7aaeeee829f77b1a1029b9f7524879bbe399
+	// outpuut: 'x' outside of string in unpack at /usr/share/perl5/vendor_perl/Image/ExifTool.pm line 5059.
+	if i := bytes.IndexByte(data, '\n'); i >= 0 {
```
Comment (barasher) [inline exiftool.go:280]: Should be unit-tested

File: exiftool.go [hunk 3]
```diff
@@ -269,16 +269,29 @@ func (e *Exiftool) WriteMetadata(fileMetadata []FileMetadata) {
...
+		if !bytes.Equal(data[:i], []byte{91, 123}) {
+			return i + 1, data[0:i], nil
```
Comment (barasher) [inline exiftool.go:282]: What are `[]byte{91, 123}` ? It's not really clear

**Review by barasher — CHANGES_REQUESTED** [review]
Could you please share a file that `go-exiftool` does not handle well ?

**Comment by barasher** [comment]
@switch-li, could you please have a look at my questions ?

**Pull request closed without merging**